hal_st/cortex/InterruptCortex.cpp

```diff
@@ -58,35 +60,36 @@ namespace hal
         : irq(other.irq)
     {
         InterruptTable::Instance().TakeOverHandler(irq, *this, other);
-        other.irq = static_cast<IRQn_Type>(0xffff);
+        other.irq = invalidIrq;
     }
 
     InterruptHandler& InterruptHandler::operator=(InterruptHandler&& other)
     {
+        InterruptTable::Instance().DeregisterHandler(irq, *this);
```

Comment:
Please either check if (irq != invalidIrq), or remove the check at the destructor